src/components/Dropdown/DropdownInput.tsx

```
@@ -0,0 +1,74 @@
+import { ChangeEvent, KeyboardEvent, useCallback, useContext, useRef } from 'react'
+import DropdownContext from './DropdownContext'
+import FocusContext from './FocusContext';
+import generateDropdownId from './generateDropdownId';
+import InputContext from './InputContext';
+
+/**
+ * An input component for use within a Dropdown.
+ */
+export default function DropdownInput(props: {
+  className?: string,
+  placeholder?: string,
+  onSubmit?: (value?: string) => void,
+  onFocus?: (value?: string) => void,
+  onType?: (value?: string) => void
+}) {
+  const {
+    className,
+    placeholder,
+    onSubmit,
+    onFocus,
+    onType
+  } = props;
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const { decoratedToggle, onSelect, screenReaderUUID } = useContext(DropdownContext) || {};
+  const { value = '', setValue, setLastTypedOrSubmittedValue } = useContext(InputContext) || {};
+  const { focusedIndex = -1, setFocusedIndex } = useContext(FocusContext) || {};
```

Should these compound component be use outside of the parent? If not, maybe we should make `useDropdownContext`, `useInputContext`, `useFocusContext` so it can guard the usage of compound component (throw error if React.useContext(BlahContext) is undefined)?
Otherwise, can we have default values for these fields or a predefined object pass to the corresponding createContext calls so we don't need to have checks when invoking these fields (blahFunction && blahFunction(...))?

---

updated! I'm normally a little worried about throwing an error and crashing the whole app but here I think we shouldn't run into any issues doing so

---

I don't think we want to provide defaults since that might cover up developer errors